interface/reports/patient_list_creation.php

```diff
@@ -246,582 +319,773 @@ function srch_option_change(elem) {
+                                            </span>
+                                            <span id="com_pref" style="display: none">
+                                                <select class="form-control" name="communication" id="communication" title="<?php echo xlt('Select Communication Preferences'); ?>">
+                                                    <option><?php echo xlt('All'); ?></option>
+                                                    <option value="allow_sms" <?php echo ($communication == "allow_sms") ? "selected" : ""; ?>><?php echo xlt('Allow SMS'); ?></option>
+                                                    <option value="allow_voice" <?php echo ($communication == "allow_voice") ? "selected" : ""; ?>><?php echo xlt('Allow Voice Message'); ?></option>
+                                                    <option value="allow_mail" <?php echo ($communication == "allow_mail") ? "selected" : ""; ?>><?php echo xlt('Allow Mail Message'); ?></option>
+                                                    <option value="allow_email" <?php echo ($communication == "allow_email") ? "selected" : ""; ?>><?php echo xlt('Allow Email'); ?></option>
+                                                </select>
+                                            </span>
+                                            <span id="ins_co" style="display: none">
+                                                <select class="form-control" name="insurance_companies" id="insurance_companies" title="<?php echo xlt('Select Insurance Company'); ?>">
+                                                    <option><?php echo xlt('All'); ?></option>
+                                                    <?php foreach ($insarr as $ins_id => $ins_co) { ?>
+                                                        <option <?php echo (!empty($_POST['insurance_companies']) && ($_POST['insurance_companies'] == $ins_id)) ? 'selected' : ''; ?> value="<?php echo $ins_id; ?>"><?php echo text($ins_co); ?></option>
+                                                    <?php } ?>
+                                                </select>
+                                            </span>
+                                            <span id="enc_type" style="display: none">
+                                                <select class="form-control" name="encounter_type" id="encounter_type">
+                                                    <option><?php echo xlt('All'); ?></option>
+                                                    <?php foreach ($encarr as $enc_id => $enc_t) { ?>
+                                                        <option <?php echo (!empty($_POST['encounter_type']) && ($_POST['encounter_type'] == $enc_id)) ? 'selected' : ''; ?> value="<?php echo $enc_id; ?>"><?php echo text($enc_t); ?></option>
+                                                    <?php } ?>
```

Comment:
$enc_id (value) needs to be wrapped with attr() function for html escaping (prevents cross-scripting attacks)